zeroconf/_core.py

```diff
@@ -834,6 +834,11 @@ def _async_send_transport(
             out,
             packet,
         )
+        # Get flowinfo and scopeid for the IPV6 socket to create a complete IPv6
+        # address tuple: https://docs.python.org/3.6/library/socket.html#socket-families
+        if s.family == socket.AF_INET6 and not v6_flow_scope:
+            _, _, sock_flowinfo, sock_scopeid = s.getsockname()
+            v6_flow_scope = (sock_flowinfo, sock_scopeid)
         transport.sendto(packet, (real_addr, port or _MDNS_PORT, *v6_flow_scope))
```

Comment:
```suggestion
        if v6_flow_scope:
	        transport.sendto(packet, (real_addr, port or _MDNS_PORT, *v6_flow_scope))
	    else:
	        transport.sendto(packet, (real_addr, port or _MDNS_PORT))
```

The idea was that we matched the source v6_flow_scope when sending the reply. If you fetch it from `s`, it is not always going to be the same as the one received.

In this case, I think we want to omit `v6_flow_scope` if its an empty response as its really only used for unicast resposnes.

Reply:
I tried the suggested change on my Windows Server 2019 machine. Unfortunately it still fails to discover services over IPv6. The services are also advertised using Zeroconf.

My change will only update `v6_flow_scope` if it wasn't provided by the source.

Reply:
I was hoping for a less risky change.  I'll do some testing on mac and linux later today.

Reply:
Mac looks good

Reply:
2nd mac looks good

Reply:
Testing passed on a home assistant container with ipv6 on linux